core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -502,6 +522,8 @@ private[spark] class BlockManagerInfo(
 
   def blocks: JHashMap[BlockId, BlockStatus] = _blocks
 
+  def cachedBlocks: mutable.HashSet[BlockId] = _cachedBlocks
```

Comment:
need to add an explicit comment here: `// This does not include broadcast blocks because ...`